extensions/AzureAISearch/AzureAISearch/AzureAISearchMemory.cs

```diff
@@ -353,21 +353,28 @@ await foreach (SearchIndex? searchIndex in indexesAsync.ConfigureAwait(false))
         return false;
     }
 
-    private async IAsyncEnumerable<string> UpsertBatchAsync(
+    public async IAsyncEnumerable<string> UpsertBatchAsync(
```

Comment:
nit: The Azure.Search.Documents package already comes with [SearchIndexingBufferedSender](/Azure/azure-sdk-for-net/blob/main/sdk/search/Azure.Search.Documents/src/Batching/SearchIndexingBufferedSender.cs).